packages/typespec-azure-resource-manager/lib/common-types/managed-identity.tsp

```diff
@@ -33,10 +33,23 @@ model ManagedServiceIdentity {
   type: ManagedServiceIdentityType;
 
   @doc("The identities assigned to this resource by the user.")
-  userAssignedIdentities?: UserAssignedIdentities;
+  userAssignedIdentities?: Record<UserAssignedIdentity>;
 }
 
 /** The set of user assigned identities associated with the resource. The userAssignedIdentities dictionary keys will be ARM resource ids in the form: '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}. The dictionary values can be empty objects ({}) in requests.", */
+@armCommonDefinition(
+  "UserAssignedIdentities",
+  {
+    version: Azure.ResourceManager.CommonTypes.Versions.v4,
+    isDefault: true,
+  },
+  "managedidentity.json"
+)
+@armCommonDefinition(
+  "UserAssignedIdentities",
+  Azure.ResourceManager.CommonTypes.Versions.v5,
+  "managedidentity.json"
+)
 model UserAssignedIdentities is Record<UserAssignedIdentity>;
```

Comment:
feel like this needs to be removed or it won't have the same meaning

Reply:
There are existing APIs that just use this common type, and we want to allow them not to break.  We would also like to generate the common types.

Reply:
If they use this type they will break when converting to typespec generation

Reply:
This should either be removed and added just for the common types generation or it should be an alias and some magic done in the common type generation

Reply:
I added as alias in a different file.

Reply:
Another option is to leave it as model but add a deprecation on it. This way the common type swagger code gen is not as hacky. I actually like it slightly more than alias. Just slightly. :-)

Reply:
@timotheeguerin  Yes, we want to have a conversation with anyone who is converting one of these specs that uses this type, so that they mave to managedserviceidentity, or use Record<UserAssignedIdentity> instead.  Meanwhile, we can use this type to generate common-types.